server/src/main/java/org/apache/druid/server/QueryCapacityExceededException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.QueryException;
+
+public class QueryCapacityExceededException extends QueryException
+{
+  private static final String ERROR_CLASS = QueryCapacityExceededException.class.getName();
+  public static final String ERROR_CODE = "Query capacity exceeded";
+  public static final String ERROR_MESSAGE = "Total query capacity exceeded";
+  public static final String ERROR_MESSAGE_TEMPLATE = "Query capacity exceeded for lane %s";
```

Comment:
Maybe worth to say what configuration is related to this error. See https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/Groupers.java#L39 as an example.

Reply:
This seems like maybe a good idea, but I am also afraid it might be a little chatty. I think people would expect to see these error messages if they configure limits, especially since there are no default limits. I think is maybe a bit different than the message you referenced, which I think is more likely something that would be bumped into by accident while trying to query something too big. Maybe if i document the error responses to expect in the configuration documentation that would be useful enough?

Reply:
That sounds good to me.

Reply:
Added docs to `querying.md` with the other error response documentation